### PR TITLE
yamivpp: color balance setting

### PR DESCRIPTION
**Pull request by wudping**
add setting for hue, saturation, brightness and contrast.

Signed-off-by: wudping <dongpingx.wu@intel.com>

### CODE REVIEW

**Comment by wudping** [comment]
corresponding libyami-utils:
[https://github.com/01org/libyami-utils/pull/57](libyami-utils)

File: tests/vpp.cpp [hunk 1]
```diff
@@ -29,6 +29,11 @@
 
 using namespace YamiMediaCodec;
 
+#define HUE_DEFAULT 50
```
Comment (xuguangxin) [inline tests/vpp.cpp]: why we use this as default value, is it special? what if I just want set hue to 50?
Could we just use COLORBALANCE_NONE etc.

File: tests/vpp.cpp [hunk 2]
```diff
@@ -179,6 +192,58 @@ class VppTest
                 case 3:
                     m_deinterlaceMode = optarg;
                     break;
+                case 4:
+                    m_hue = atoi(optarg);
+                    if (m_hue > COLORBALANCE_LEVEL_MAX) {
```
Comment (xuguangxin) [inline tests/vpp.cpp]: this check is not needed, we should let the library check this

File: tests/vpp.cpp [hunk 3]
```diff
@@ -246,6 +311,58 @@ class VppTest
                 return false;
             }
         }
+
+        if (m_hue != HUE_DEFAULT) {
+            VPPColorBalanceParameters hue;
```
Comment (xuguangxin) [inline tests/vpp.cpp]: so many duplicated code, could we use a function for this?

File: tests/vpp.cpp [hunk 4]
```diff
@@ -29,6 +29,8 @@
 
 using namespace YamiMediaCodec;
 
+#define COLOR_BALANCE_DEFAULT -1
```
Comment (xuguangxin) [inline tests/vpp.cpp]: Most ok to me.But this should be COLORBALANCE_LEVEL_NONE. currently it's same value -1, but in future we may change COLORBALANCE_LEVEL_NONE to other value, so use COLORBALANCE_LEVEL_NONE for default is better

Comment (wudping) [inline tests/vpp.cpp]: okay， thanks @xuguangxin .
#define COLOR_BALANCE_DEFAULT COLORBALANCE_LEVEL_NONE 
according to your suggestion, I hope the above piece code can work.

**Comment by xuguangxin** [comment]
verify +1

**Comment by xuguangxin** [comment]
merged as 6f70d370e2c9605d1592762ad8c238c4e74514ac

**Pull request closed without merging**